low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -97,27 +98,26 @@ def get_controllers_info(self):
         data = []
         controllers = self.get_realstor_encl_data("controllers")
         for controller in controllers:
-            controller_dict = {
-              "uid": controller.get("durable-id"),
-              "fru": "true",
-              "last_updated": int(time.time()),
-              "health": {
-                "status": controller.get("health", "NA"),
-                "description": controller.get("description"),
-                "recommendation": controller.get("health-recommendation"),
-                "specifics": [
-                  {
+            uid = controller.get("durable-id")
+            status = controller.get("health", "NA")
+            description = controller.get("description")
+            recommendation = controller.get("health-recommendation")
+            controller_dict = self.get_health_template(uid, is_fru=True)
+            controller_dict["last_updated"] = int(time.time())
```

Comment:
get_health_template() can set `"last_updated"` attribute. So this line of work can be avoided in all FRUs method.

Reply:
Done

Reply:
as suggested by Chetan, we should update the last_upated time once we have collected the data, so it should be in `set_health_data()` function instead of `get_health_template()`.

Reply:
I agree.

Reply:
Done, Added 'last_updated' field in set_health_data function.